### DISCUSSION TITLE
FormItem 自定义控件 onChange 需要放上 event 还是放上值?

### DISCUSSION THREAD

**Discussion by vaynevayne** [ant-design/ant-design#43009] [Q&A]
https://ant.design/components/form-cn#components-form-demo-customized-form-controls

这个 demo 里的 onChange 是直接放入值, 
但是 antd Input组件的 onCHange 是放入 event, 这都能工作,
请问黑盒子里的逻辑是什么?

**Comment by linxianxi** [comment]
都可以，event 配合 `getValueFromEvent ` 使用即可

**Reply by vaynevayne** [reply]
我说的是antd内部

**Reply by linxianxi** [reply]
你的标题不是问的自定义？

https://github.com/react-component/field-form/blob/master/src/Field.tsx#L562

**Reply by vaynevayne** [reply]
请问黑盒子里的逻辑是什么?

**Reply by linxianxi** [reply]
配合上面那个链接食用   https://github.com/react-component/field-form/blob/master/src/utils/valueUtil.ts#LL73C17-L73C41

**Reply by vaynevayne** [reply]
你的意思是需要用户自己加上这个 prop 吗?  
那就奇怪了, 我没有加, onChange 时放上 event 也能跑

**Reply by linxianxi** [reply]
发的链接源码不是写了吗？ 用户没传 getValueFromEvent 时会走 defaultGetValueFromEvent 逻辑。defaultGetValueFromEvent 里如果有 event.target.value 就会用它
